delta/ml/predict.py

```diff
@@ -181,7 +216,8 @@ def predict(self, image, label=None, input_bounds=None, overlap=(0, 0)):
         if net_input_shape[0] is None and net_input_shape[1] is None:
             assert net_output_shape[0] is None and net_output_shape[1] is None
             out_shape = self._model.compute_output_shape((0, ts[0], ts[1], net_input_shape[2]))
-            tiles = input_bounds.make_tile_rois(ts, include_partials=False,
```

Comment:
I think you want include_partials=False, partials_overlap will not be applied if it is true. Setting include_partials will make all the tiles not be the same shape

Reply:
I think something is wrong with the tile logic here, if I set that to false some of my input shapes are skipped.  For example:
input_bounds = min_x: 643, max_x: 868, min_y: 1464, max_y: 1538
tiles = []

Reply:
what is your tile shape? If it's larger than the input_bounds it would correctly skip it. With include_partials=False I believe it will crash sometimes with fully convolutional networks (the ones with shape (~, ~, num_bands).

Reply:
So what happens if I want to classify an image that does not evenly fit my tile size?  Are the extra portion of the image just discarded?  This issue greatly affects manually specified regions but it also applies to the whole image classification.

Reply:
That's a good point. But the issue is, most of the networks have a minimum size. If they don't have data of that size they just won't work. I guess for the manually specified regions, we could give a bigger minimum sized region and only count the part in the rectangle. What do you think @mvonpohle?

For the extra portions, the partials_overlap will make multiple tiles of the correct size which completely cover the image but overlap. Thinking about it more, this works for constructing the output image, but we may be double counting the overlapping parts in our metrics...

Reply:
I moved more overlap functionality to the overlap class, so far it seems to be handling everything.